assets/dev/js/editor/utils/localized-value-store.js

```diff
@@ -0,0 +1,39 @@
+export default class LocalizedValueStore {
+	constructor() {
+		this.store = [];
+	}
+
+	sendKey( event ) {
+		const isLetter = event.keyCode >= 65 && event.keyCode <= 90;
+		const isSpace = 32 === event.keyCode || ' ' === event.originalEvent.data;
+
+		if ( isLetter || isSpace ) {
+			this.store.push( { original: event.originalEvent.key, english: String.fromCharCode( event.keyCode ) } );
+		}
+		if ( ! event.target.value || event.target.value.length < this.store.length ) {
+			this.store = this.rebuildStore( event.target.value );
+		}
+	}
+
+	shouldBeIgnored( event ) {
+		return event.shiftKey || event.ctrlKey || event.altKey;
+	}
+
+	getLocalizedValue() {
+		return this.store.map( ( x ) => x.english ).join( '' );
+	}
+
+	rebuildStore( value ) {
+		const chars = value.split( '' );
+		const newStore = [];
+		let charsIndex = 0;
+		for ( let storeIndex = 0; storeIndex < this.store.length; storeIndex++ ) {
+			if ( this.store[ storeIndex ].original === chars[ charsIndex ] ) {
+				newStore.push( this.store[ storeIndex ] );
+				charsIndex++;
+			}
+		}
+
+		return newStore;
```

Comment:
Maybe something like this?

```suggestion
		return chars.map( ( char ) => this.localizeFromCache( char ) );
```

Reply:
i agree that its not 100% clean code , i will refactor it , it uses 2 indices storeIndex & charsIndex to find if the string was changed by deletion